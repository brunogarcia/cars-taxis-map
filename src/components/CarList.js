import React from 'react';
import PropTypes from 'prop-types';
import CarItem from './CarItem';
import Types from '../utils/types';

const CarList = props => props.data.map(item => <CarItem key={item.id} data={item} />);

CarList.propTypes = {
  data: PropTypes.arrayOf(Types.car.isRequired).isRequired,
};

export default CarList;
