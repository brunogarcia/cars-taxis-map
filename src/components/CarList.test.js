import React from 'react';
import renderer from 'react-test-renderer';
import CarList from './CarList';

test('renders correctly', () => {
  const props = {
    data: [
      {
        address: 'test 1',
        coordinates: [24, 81, 0],
        engineType: 'test',
        exterior: 'test',
        fuel: 99,
        interior: 'test',
        name: 'test',
        vin: 'test',
        id: 12345,
      },
      {
        address: 'test 2',
        coordinates: [41, 41, 0],
        engineType: 'test',
        exterior: 'test',
        fuel: 32,
        interior: 'test',
        name: 'test',
        vin: 'test',
        id: 56789,
      },
    ],
  };

  const component = renderer.create(<CarList {...props} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
