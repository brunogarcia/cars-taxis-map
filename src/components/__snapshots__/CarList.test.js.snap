// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1 CardItem-main"
>
    <div
        className="MuiCardContent-root-29"
    >
        <h2
            className="MuiTypography-root-30 MuiTypography-headline-35"
        >
            test 1
        </h2>
        <div
            className="CardItem-fuel"
        >
            <span
                className="MuiBadge-root-54"
            >
                <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-59 Battery-full"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                >
                    <g>
                        <path
                            d="M15.67 4H14V2h-4v2H8.33C7.6 4 7 4.6 7 5.33v15.33C7 21.4 7.6 22 8.33 22h7.33c.74 0 1.34-.6 1.34-1.33V5.33C17 4.6 16.4 4 15.67 4zM11 20v-5.5H9L13 7v5.5h2L11 20z"
                        />
                    </g>
                </svg>
                <span
                    className="MuiBadge-badge-55 MuiBadge-colorPrimary-56"
                >
                    99
                </span>
            </span>
        </div>
        <p
            className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-52"
        >
            Plate: 
            test
        </p>
        <p
            className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-52"
        >
            Latitude: 
            24
             
            <br />
            Longitude: 
            81
        </p>
    </div>
</div>,
  <div
    className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1 CardItem-main"
>
    <div
        className="MuiCardContent-root-29"
    >
        <h2
            className="MuiTypography-root-30 MuiTypography-headline-35"
        >
            test 2
        </h2>
        <div
            className="CardItem-fuel"
        >
            <span
                className="MuiBadge-root-54"
            >
                <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-59"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                >
                    <g>
                        <path
                            d="M14.47 13.5L11 20v-5.5H9l.53-1H7v7.17C7 21.4 7.6 22 8.33 22h7.33c.74 0 1.34-.6 1.34-1.33V13.5h-2.53z"
                        />
                        <path
                            d="M15.67 4H14V2h-4v2H8.33C7.6 4 7 4.6 7 5.33v8.17h2.53L13 7v5.5h2l-.53 1H17V5.33C17 4.6 16.4 4 15.67 4z"
                            fillOpacity=".3"
                        />
                    </g>
                </svg>
                <span
                    className="MuiBadge-badge-55 MuiBadge-colorPrimary-56"
                >
                    32
                </span>
            </span>
        </div>
        <p
            className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-52"
        >
            Plate: 
            test
        </p>
        <p
            className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-52"
        >
            Latitude: 
            41
             
            <br />
            Longitude: 
            41
        </p>
    </div>
</div>,
]
`;
